### ISSUE TITLE
No-slip landing

### ISSUE THREAD

**Issue by Regulator13** [famegames/fire-and-ice#27]
**Context** 
Having easy to navigate controls and functionality is important in a game of reflexes and timing, especially when lag is involved.

**Problem or idea**
Players slip off of blocks when they land on them due to greater horizontal speeds than friction can slow compensate for, leading to players using the old 'turn around and land on the block from the opposite direction'.

**Solution or next step**
I think this needs to be handled carefully so we don't mess up the beauty of sliding blocks (which I like). If we can just increase player friction that's fine. Perhaps something more complex like subtracting vertical speed from horizontal speed when landing then applying friction so high arc jumps are easy to land but long jumps are still difficult.

*Include any tags that apply. Tag a person if you are looking for a response from someone specific.*
@Regulator13 - I would like to hear how this problem is addressed, if someone else besides me does address it.

**Comment by afrey15** [comment]
That's a pretty keen observation, I like the way you're thinking. I like the second 'complex' solution and I'll add another thought. How about a dramatic reduction in the player's ability to change direction in midair? I'm thinking rather than pressing the opposite directional arrow to change directions when in the air, we could make it slow the player's horizontal speed. Only once the number reaches 0 will they change directions. Ideally, this would prevent directional switching in all jumps and allow it pretty much only when the player is falling

**Comment by Regulator13** [comment]
This is an interesting idea but I don't see it as fitting for a platformer, partially because of how hard it would make the game. Imagine you have a ledge like this where you're on the lower layer
========
YOU
========
Being able to jump to the second layer is important, as this happens frequently. Without it you'd need a block. 

Mostly I think it's more fun to be able to nimbly move around

**Comment by Phabbits** [comment]
That is true, and as you found out the issue of slipping off the platforms had already been addressed before uploading. What I was thinking might be a helpful feature, especially for new players, is adding in a functionality where if you hit the top half of the block horizontally, the character will automatically climb to the top. This should make it much easier to hit blocks out in space while not deterring too much for the game play. Remember, the game should be difficult not annoying. I believe that the difficulty can be achieved from much more fun things. Finally I could see a possibility of adding a fast drop functionality, either as an item or built in, that would let you descend quicker and not over jump blocks.

**Comment by Regulator13** [comment]
I had forgotten about the climb feature when hitting the side of frozen blocks, I will make an issue for this. To address @afrey15, I agree it might _look_ cleaner but I think it will be substantially harder and more frustrating to play. I don't know what theme you think we're moving towards but maybe you'd like to share it with the group in a discussion thread. 

> If blocks are stacked with no overhang you already have two options to proceed. We don't need to change directions in midair to be able to pass these obstacles.

Also you still didn't say what would happen if blocks _are_ stacked with overhang or what these two options are.

**Comment by Regulator13** [comment]
I've submitted a pull request #49 that should fix this issue with the struggle/climb method. Test it out, if it's acceptable I'm going to close this issue
